sql: update EXPLAIN related document (#1039)
pingcap/docs: sql/understanding-the-query-execution-plan.md

- `EXPLAIN` works together with `SELECT`, `DELETE`, `INSERT`, `REPLACE`, and `UPDATE`.
- When you run the `EXPLAIN` statement, TiDB returns the final physical execution plan which is optimized by the SQL statement of `EXPLAIN`. In other words, `EXPLAIN` displays the complete information about how TiDB executes the SQL statement, such as in which order, how tables are joined, and what the expression tree looks like. For more information, see [`EXPLAIN` output format](#explain-output-format).

> - TiDB now supports `EXPLAIN [options] FOR CONNECTION connection_id` which works differently than `EXPLAIN FOR` in MySQL. See [`EXPLAIN FOR CONNECTION`](#explain-for-connection)。

Reviewer: ```suggestion
- TiDB also supports `EXPLAIN [options] FOR CONNECTION connection_id`, but with minor differences from MySQL. See [`EXPLAIN FOR CONNECTION`](#explain-for-connection).
```

- I don't think users need the history that TiDB did not support EXPLAIN FOR CONNECTION.
- I think the semantic differences are only minor.  We should not scare users it is completely different, given that we have already explained the output format of EXPLAIN is different.

+------------------------+---------+------+--------------------------------------------------------------------------------------------------+-----------------------------------+
4 rows in set (0.08 sec)
```
## `EXPLAIN FOR CONNECTION`

> `EXPLAIN FOR CONNECTION` is used for getting the last query execution plan in one connection. The output format is totally the same as `EXPLAIN`. But the execution in TiDB is different from that in MySQL. Except for the output format, there are other differences as followed: 

Reviewer: ```suggestion
`EXPLAIN FOR CONNECTION` provides the last query execution plan used by a given connection. The output format is the same as `EXPLAIN` in TiDB. The usage has the following semantic differences from MySQL: 
```

> - **EXECUTE** query plan is returned in MySQL while **LAST EXECUTED** is returned in TiDB.

Reviewer: ```suggestion
- MySQL returns the plan for the currently **executing query** while TiDB returns the execution plan for the **last executed** query.
```

> - In MySQL document, the connection of the user account should be the same as that of query or the user account has the `PROCESS` privileges. For TiDB, the user account should have the same connection as the query or has the `SUPER` privileges.   

Reviewer: ```suggestion
- In MySQL, executing `EXPLAIN FOR CONNECTION` for connections owned by other users requires the `PROCESS` privilege. In TiDB, this statement requires the `SUPER` privilege.
```
